Steer/DigitizerWorkflow/src/FDDDigitWriterSpec.h

```diff
@@ -27,17 +27,26 @@ namespace fdd
 template <typename T>
 using BranchDefinition = framework::MakeRootTreeWriterSpec::BranchDefinition<T>;
 
-o2::framework::DataProcessorSpec getFDDDigitWriterSpec()
+o2::framework::DataProcessorSpec getFDDDigitWriterSpec(bool mctruth = true)
 {
   using InputSpec = framework::InputSpec;
   using MakeRootTreeWriterSpec = framework::MakeRootTreeWriterSpec;
-  return MakeRootTreeWriterSpec("FDDDigitWriter",
-                                "fdddigits.root",
-                                "o2sim",
-                                1,
-                                BranchDefinition<std::vector<o2::fdd::Digit>>{InputSpec{"digitBCinput", "FDD", "DIGITSBC"}, "FDDDigit"},
-                                BranchDefinition<std::vector<o2::fdd::ChannelData>>{InputSpec{"digitChinput", "FDD", "DIGITSCH"}, "FDDDigitCh"},
-                                BranchDefinition<o2::dataformats::MCTruthContainer<o2::fdd::MCLabel>>{InputSpec{"labelinput", "FDD", "DIGITLBL"}, "FDDDigitLabels"})();
+  if (mctruth) {
+    return MakeRootTreeWriterSpec("FDDDigitWriter",
+                                  "fdddigits.root",
+                                  "o2sim",
+                                  1,
+                                  BranchDefinition<std::vector<o2::fdd::Digit>>{InputSpec{"digitBCinput", "FDD", "DIGITSBC"}, "FDDDigit"},
+                                  BranchDefinition<std::vector<o2::fdd::ChannelData>>{InputSpec{"digitChinput", "FDD", "DIGITSCH"}, "FDDDigitCh"},
+                                  BranchDefinition<o2::dataformats::MCTruthContainer<o2::fdd::MCLabel>>{InputSpec{"labelinput", "FDD", "DIGITLBL"}, "FDDDigitLabels"})();
+  } else {
+    return MakeRootTreeWriterSpec("FDDDigitWriter",
+                                  "fdddigits.root",
+                                  "o2sim",
+                                  1,
+                                  BranchDefinition<std::vector<o2::fdd::Digit>>{InputSpec{"digitBCinput", "FDD", "DIGITSBC"}, "FDDDigit"},
+                                  BranchDefinition<std::vector<o2::fdd::ChannelData>>{InputSpec{"digitChinput", "FDD", "DIGITSCH"}, "FDDDigitCh"})();
+  }
```

Comment:
@matthiasrichter : I wondering if there is a shorter way here to configure the list of BranchDefinitions based on an external parameter (essentially to influence the list of variadic arguments)?

Reply:
I assume your question goes in the direction I have answered above?

Reply:
below :-)